### ISSUE TITLE
Need documentation for bundling ES6 Leaflet modules

### ISSUE THREAD

**Issue by piehei** [Leaflet/Leaflet#5807]
Since the transformation to ES6 it should be possible to import only the needed parts of Leaflet and not the whole library.

Is there any documentation, examples, or whatnot, available for this kind of usage?

It would be extremely great to have a simple guide on this which would list inter-dependencies, what you should note when importing just a piece, and so forth. Like "to have this and that functionality, import this" and "if you import that, also this is going to get imported since that depends on it".

Any thoughts?

**Comment by Eschon** [comment]
👍 

I think this would be incredibly usefull. Right now leaflet takes up about 1/3 of my bundle size. Since I only use a fraction of the functionality this could be reduced a lot by only importing the parts that I need

**Comment by IvanSanchez** [comment]
I was tempted to close this as a duplicate of https://github.com/Leaflet/Leaflet/issues/5696 , but then I noticed that this is actually a *request for documentation* on how to bundle ES6 Leaflet.

**Comment by piehei** [comment]
@IvanSanchez yes I read through that and thought this doesn't belong to the issue thread. Should have noted in my original msg :)

**Comment by vishnugaruda246** [comment]
@IvanSanchez have a review

**Comment by vishnugaruda246** [comment]
@IvanSanchez any updates?

**Comment by jonkoops** [comment]
For this issue what we want is a very small piece of documentation on the [download page](https://leafletjs.com/download.html). Perhaps we should change the terminology on this page to something else as well, but that can be handled separately.

This page must include must include some simple instructions on how to install Leaflet using NPM (already have this), and how to integrate it in an environment of a bundler (e.g. WebPack, Vite, etc.), preferably without naming any specific bundling tools. This could be something like how to import the Leaflet code and how to include the styling on the page.
